modules/tests/test_ref_modules/test_ref_modules.cpp

```diff
@@ -575,5 +575,493 @@ TEST_F(RefModulesTest, UpdateDevicePower)
+            ASSERT_EQ(static_cast<Float>(mark), expectedMark);
+            expectedMark += classCount;
+        }
+        ASSERT_EQ(labels[labels.getCount() - 1], inputRange.getHighValue());
+    }
+
+    const auto classifierDomainSignal = classifierSignal.getDomainSignal();
+    const auto classifierDomainSignalDescription = classifierDomainSignal.getDescriptor();
+
+    const auto deviceDomainDescriptor = deviceSignal.getDomainSignal().getDescriptor();
+
+    const auto deviceDomainRule = deviceDomainDescriptor.getRule();
+    const auto classifierDomainRule = classifierDomainSignalDescription.getRule();
+
+    // Check Linear Rule
+    if (deviceDomainRule.getType() == DataRuleType::Linear)
+    {
+        ASSERT_EQ(deviceDomainRule.getType(), classifierDomainRule.getType());
+        ASSERT_EQ(classifierDomainSignalDescription.getTickResolution(), deviceDomainDescriptor.getTickResolution());
+
+        auto deviceResolution = deviceDomainDescriptor.getTickResolution() * Int(1000);
+
+        auto classifierDelta = classifierDomainRule.getParameters().get("delta");
+        auto expectedClassifierDelata = static_cast<Float>(blockSize) / static_cast<Float>(deviceResolution);
+
+        ASSERT_EQ(classifierDelta, expectedClassifierDelata);
+    }
+    else
+    {
+        ASSERT_EQ(classifierDomainRule.getType(), DataRuleType::Explicit);
+    }
+}
+
+TEST_F(RefModulesTest, ClassifierCheckBuildinClassCustomRange)
+{
+    const auto instance = Instance();
+
+    const auto device = instance.addDevice("daqref://device0");
+
+    auto customClassList = List<Float>(1, 10, 100, 1000);
+    auto blockSize = 2;
+    auto classCount = 2;
+    auto inputRange = Range(-20.0, 20.0);
+    auto classifierInputRange = Range(0.0, 20.0);
+
+    const auto classifierFb = instance.addFunctionBlock("ref_fb_module_classifier");
+    classifierFb.setPropertyValue("UseCustomClasses", false);
+    classifierFb.setPropertyValue("CustomClassList", customClassList);
+    classifierFb.setPropertyValue("BlockSize", blockSize);
+    classifierFb.setPropertyValue("ClassCount", classCount);
+    classifierFb.setPropertyValue("BlockSize", blockSize);
+    classifierFb.setPropertyValue("UseCustomInputRange", true);
+    classifierFb.setPropertyValue("inputLowValue", classifierInputRange.getLowValue());
+    classifierFb.setPropertyValue("InputHighValue", classifierInputRange.getHighValue());
+
+    const auto deviceChannel = device.getChannels()[0];
+    deviceChannel.setPropertyValue("CustomRange", inputRange);
+    const auto deviceSignal = deviceChannel.getSignals()[0];
+
+    const auto classifierInputPort = classifierFb.getInputPorts()[0];
+    classifierInputPort.connect(deviceSignal);
+
+    const auto classifierSignal = classifierFb.getSignals()[0];
+    const auto classifierSignalDescription = classifierSignal.getDescriptor();
+
+    // Classifier returning values with float float
+    ASSERT_EQ(classifierSignalDescription.getSampleType(), SampleType::Float64);
```

Comment:
Same as above, these tests are repeated and somewhat obfuscate the purpose of the test.

The goal of each test should be clear in what it's testing. If the test is titled "CheckBuiltInClassCustomRange" it should test for that, not for all other unrelated properties.

In general, these tests should aim to be as short as possible. Most of the tests available here should be split into 2-4 smaller, more targeted test cases.